mongo_connector/doc_managers/neo4j_doc_manager.py

```diff
@@ -0,0 +1,85 @@
+"""
+Neo4j implementation for the DocManager. Receives documents and 
+communicates with Neo4j Server.
+"""
+import base64
+import logging
+
+import os.path as path, sys
+
+from threading import Timer
+
+import bson.json_util
+
+from py2neo import Graph, Node, Relationship, watch
```

Comment:
Please don't use Node and Relationship afaik they use the very old and slow REST API

use (concurrent) cypher with the transactional API
you can insert hundreds of thousands of nodes and rels per second
you can send up to 1000 statements per request and up to 50-100k nodes/rels per transaction.
See: http://py2neo.org/2.0/cypher.html#transactions